### PR TITLE
Temp elemental fireballs + lightning bolt

### PR DESCRIPTION
**Pull request by Thorvers**

### CODE REVIEW

File: RPG Wizardry/Assets/Scripts/Sorcery/Spells/LightningBolt.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                lineRenderer.SetPosition(0, transform.localPosition);
+                lineRenderer.SetPosition(1, transform.position + transform.up * data.LifeTime);
+                Debug.DrawRay(transform.position, transform.up * data.LifeTime, Color.red,5);
+                Debug.Log("Did not Hit");
+            }
+            //Animate the line
+            StartCoroutine(CreateBolt());
+        }
+
+        //Displays line, then deletes self
```
Comment (Sven-dev) [inline RPG Wizardry/Assets/Scripts/Sorcery/Spells/LightningBolt.cs]: Make a summary of this

**Review by Sven-dev — APPROVED** [review]

File: RPG Wizardry/Assets/Scripts/Sorcery/Spells/LightningBolt.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            hit = Physics2D.Raycast(transform.position, transform.up, data.LifeTime, collisionLayer);
+            if (hit)
+            {
+                //If it's an object with health, damage it
+                hit.transform.GetComponent<IHealth>()?.Damage(data.Damage);
+                //Set the line to end at the object
+                lineRenderer.SetPosition(0, transform.localPosition);
+                lineRenderer.SetPosition(1, hit.point);
+                Debug.DrawRay(transform.position, transform.up * hit.distance, Color.yellow, 5);
+                Debug.Log("Did Hit");
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/Sorcery/Spells/LightningBolt.cs]: Remove Debug Logs & Rays before merging

File: RPG Wizardry/Assets/Scripts/Sorcery/Spells/LightningBolt.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            }
+            //No, it didn't hit anything
+            else
+            {
+                //Set the line to end at the max distance
+                lineRenderer.SetPosition(0, transform.localPosition);
+                lineRenderer.SetPosition(1, transform.position + transform.up * data.LifeTime);
+                Debug.DrawRay(transform.position, transform.up * data.LifeTime, Color.red,5);
+                Debug.Log("Did not Hit");
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/Sorcery/Spells/LightningBolt.cs]: See above

File: RPG Wizardry/Assets/Scripts/Sorcery/Spells/LightningBolt.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            //Animate the line
+            StartCoroutine(CreateBolt());
+        }
+
+        //Displays line, then deletes self
+        private IEnumerator CreateBolt()
+        {
+            //Set line to visible
+            lineRenderer.enabled = true;
+            yield return new WaitForSeconds(0.1f);
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/Sorcery/Spells/LightningBolt.cs]: This shouldn't be a coroutine. Use Destroy(obj, t) instead
https://docs.unity3d.com/ScriptReference/Object.Destroy.html

**Review by frankvHoof93 — APPROVED** [review]

**Review by Sanko-Kallig — APPROVED** [review]

**Pull request merged**